lib/provider.js

```diff
@@ -204,11 +204,19 @@ export default function() {
                 $log.debug('[native transition] same state transition are not possible');
                 return;
             }
+            var deferred = $q.defer();
             unregisterToStateChangeStartEvent();
-            var statePromise = $state.go(state, stateParams, stateOptions);
             transition(transitionOptions);
-            
-            return statePromise;
+            $timeout(function () {
```

Comment:
that's better but the `$timeout` already returns a promise. https://docs.angularjs.org/api/ng/service/$timeout

You can just do that:

```
return $timeout(() => $state.go(state, stateParams, stateOptions))
```

It will therefore return a promise that resolves the `$state.go` promise.